recomposition-logger-plugin/compiler-plugin/build.gradle.kts

```diff
@@ -1,52 +1,69 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
-  kotlin("jvm")
-  kotlin("kapt")
-  id("com.github.gmazzo.buildconfig")
-  `maven-publish`
-  id("plugin-options-config")
+    kotlin("jvm")
+    kotlin("kapt")
+    id("com.github.gmazzo.buildconfig")
+    `maven-publish`
+    `signing`
+    id("plugin-options-config")
 }
 
 group = pluginConfig.group
 version = pluginConfig.version
 
 dependencies {
-  compileOnly("org.jetbrains.kotlin:kotlin-compiler-embeddable")
+    compileOnly("org.jetbrains.kotlin:kotlin-compiler-embeddable")
 
-  compileOnly("com.google.auto.service:auto-service-annotations:1.0.1")
-  kapt("com.google.auto.service:auto-service:1.0-rc7")
+    compileOnly("com.google.auto.service:auto-service-annotations:1.0.1")
+    kapt("com.google.auto.service:auto-service:1.0-rc7")
 }
 
 buildConfig {
-  packageName(pluginConfig.group + ".compiler_plugin")
-  buildConfigField("String", "KOTLIN_PLUGIN_ID", "\"${pluginConfig.compilerPluginId}\"")
+    packageName(pluginConfig.group + ".compiler_plugin")
+    buildConfigField("String", "KOTLIN_PLUGIN_ID", "\"${pluginConfig.compilerPluginId}\"")
 }
 
 tasks.withType<KotlinCompile> {
-  kotlinOptions.jvmTarget = "11"
+    kotlinOptions.jvmTarget = "11"
 }
 
 java {
-  sourceCompatibility = JavaVersion.VERSION_11
-  targetCompatibility = JavaVersion.VERSION_11
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
+extensions.configure<SigningExtension> {
+    sign(publishing.publications)
 }
 
 publishing {
-  repositories {
-    //TODO: set public repo
-    mavenLocal()
-  }
-  publications {
-    create<MavenPublication>("default") {
-      artifactId = pluginConfig.compilerPluginName
-      from(components["java"])
-      artifact(tasks.kotlinSourcesJar)
+    setupPublishingRepositories(project)
+
+    publications {
+        create<MavenPublication>("default") {
+            artifactId = pluginConfig.compilerPluginName
+            from(components["java"])
+            artifact(tasks.kotlinSourcesJar)
+
+            pom {
+                setupDefaultPom()
+                name.set("Recomposition Logger compiler plugin")
+                description.set("Kotlin compiler plugin, that inject logs into composable functions")
+                url.set("https://github.com/welltech-com/recompose-logger")
+            }
+        }
     }
-  }
 }
 
-// TODO change for publishing to public repo
 tasks.register("buildAndPublishToMavenLocal") {
-  dependsOn(tasks.named("assemble"), tasks.named("publishToMavenLocal"))
+    dependsOn(tasks.named("assemble"), tasks.named("publishDefaultPublicationToMavenLocal"))
+}
+
+tasks.register("buildAndPublishToMavenRepository") {
+    dependsOn(tasks.named("assemble"), tasks.named("publishDefaultPublicationToMavenRepository"))
+}
+
+tasks.register("buildAndPublishToMaven2Repository") {
```

Comment:
Mave2?

Reply:
Renamed to `buildAndPublishToSnapshotRepository`